src/Neo.Extensions/StringExtensions.cs

```diff
@@ -43,5 +43,30 @@ public static int GetVarSize(this string value)
             var size = Utility.StrictUTF8.GetByteCount(value);
             return UnsafeData.GetVarSize(size) + size;
         }
+
+        /// <summary>
+        /// Compares two <see cref="string"/>s for equality in constant time.
+        /// </summary>
+        /// <param name="left">The left <see cref="string"/>.</param>
+        /// <param name="right">The right <see cref="string"/>.</param>
+        /// <returns>True if the two <see cref="string"/>s are equal, false otherwise.</returns>
+        public static bool ConstantTimeEquals(this string left, string right)
+        {
+            if (left == null && right == null)
+                return true;
+
+            if (left == null || right == null)
+                return false;
+
+            if (left.Length != right.Length)
+                return false;
```

Comment:
This is no constant, maybe we should do a dummy for for the maxLength string

Reply:
> This is no constant, maybe we should do a dummy for for the maxLength string

Implementations  in Rust `subtle` and Golang `crypto/subtle`  are similar

Reply:
> > This is no constant, maybe we should do a dummy for for the maxLength string
> 
> Implementations in Rust `subtle` and Golang `crypto/subtle` are similar to this one. Length is not sensitive.

Then this implementations are wrong 🙃